### PR TITLE
Added support for map selection

### PR DESCRIPTION
**Pull request by Lemongrass3110**
* **Addressed Issue(s)**: #6632

* **Server Mode**: Both

* **Description of Pull Request**: 
Added support for the map selection window in case a map is unavailable anymore.
Allows your users to pick a map from a list of defined maps that was previously only used to warp your players automatically to one of those maps, if available. By default you ended up in Prontera.

Thanks to @Xypr0, @Toshiro90, @alisonrag and @aleos89

### CODE REVIEW

File: src/char/char_clif.cpp [hunk 1]
```diff
@@ -816,6 +820,165 @@ void chclif_send_map_data( int fd, std::shared_ptr<struct mmo_charstatus> cd, ui
...
+	node->account_id = sd->account_id;
+	node->char_id = cd->char_id;
+	node->login_id1 = sd->login_id1;
+	node->login_id2 = sd->login_id2;
+	node->sex = sd->sex;
+	node->expiration_time = sd->expiration_time;
+	node->group_id = sd->group_id;
+	node->ip = ipl;
+
+	char_get_authdb()[node->account_id] = node;
```
Comment (vstumpf) [inline src/char/char_clif.cpp:943]: Lets move this instead of assignment

File: src/char/char_clif.cpp [hunk 2]
```diff
@@ -816,6 +820,165 @@ void chclif_send_map_data( int fd, std::shared_ptr<struct mmo_charstatus> cd, ui
...
+
+	return 1;
+#else
+	return 0;
+#endif
+}
+
+void chclif_accessible_maps( int fd ){
+#if PACKETVER >= 20100714
+	struct PACKET_HC_NOTIFY_ACCESSIBLE_MAPNAME* p = (struct PACKET_HC_NOTIFY_ACCESSIBLE_MAPNAME*)packet_buffer;
```
Comment (vstumpf) [inline src/char/char_clif.cpp:953]: If you don't want a global, you can create a static packet buffer here

**Review by vstumpf — COMMENTED** [review]
Some initial thoughts

File: src/char/char_clif.cpp (hunk 1 above)
Comment (Lemongrass3110) [reply src/char/char_clif.cpp:943]: As discussed in Discord this should be dealt with in a broader scope.
But good point!

**Review by aleos89 — APPROVED** [review]

**Pull request merged**